lambda/script.py

```diff
@@ -0,0 +1,121 @@
+import json
+import os
+
+# from pprint import pprint
+from typing import Dict, List, Optional
+
+import boto3  # type: ignore
+import googleapiclient.discovery  # type: ignore
+from google.oauth2 import service_account  # type: ignore
+
+
+def get_env_var(env) -> Optional[str]:
+    return os.environ.get(env)
+
+
+def get_scope(SCOPE: Optional[str]) -> List[str]:
+    client = boto3.client("ssm")
+    return [client.get_parameter(Name=SCOPE, WithDecryption=True)["Parameter"]["Value"]]
+
+
+def get_subject_email(SUBJECT: Optional[str]) -> str:
+    client = boto3.client("ssm")
+    return client.get_parameter(Name=SUBJECT, WithDecryption=True)["Parameter"]["Value"]
+
+
+def get_credentials_file(CREDENTIALS: Optional[str]) -> str:
+    client = boto3.client("ssm")
+    content = client.get_parameter(Name=CREDENTIALS, WithDecryption=True)["Parameter"][
+        "Value"
+    ]
+
+    with open("/tmp/credentials.json", "w") as outfile:
+        outfile.write(content)
+    return "/tmp/credentials.json"
+
+
+def create_admin_client(
+    SERVICE_ACCOUNT_FILE: str, scope: List[str], subject: str, pageToken: str = None
+):
+    client = boto3.client("ssm")
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=scope, subject=subject,
+    )
+    # https://googleapis.github.io/google-api-python-client/docs/epy/googleapiclient.discovery-pysrc.html#build
+    client = googleapiclient.discovery.build(
+        "admin", "directory_v1", credentials=credentials, cache_discovery=False
+    )
+
+    return client
+
+
+def create_groups_client(
+    SERVICE_ACCOUNT_FILE: str, scope: List[str], subject: str, pageToken: str = None
+):
+    client = boto3.client("ssm")
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=scope, subject=subject,
+    )
+
+    client = googleapiclient.discovery.build(
+        "groupssettings", "v1", credentials=credentials, cache_discovery=False
+    )
+
+    return client
+
+
+def build_group_dict() -> Dict[str, str]:
+    response = create_admin_client(
+        get_credentials_file(get_env_var("CREDENTIALS")),
+        get_scope(get_env_var("ADMIN_SCOPE")),
+        get_subject_email(get_env_var("SUBJECT")),
+    )
+    group_ids = {}
+    hasNextPageToken = True
+    nextPageToken = None
+
+    while hasNextPageToken:
+
+        groups = (
```

Comment:
Does this need to be wrapped in `()`?

Reply:
it does not

Reply:
Black would disagree with you on that one

Reply:
oh damn! why is it doing that, too long?

Reply:
if you don't want the enclosing brackets (which is a bit odd, no human would intentionally do that), perhaps split up the groups, the list, and the execute?

Reply:
I'm happy for black to do its thing this with this, non-issue for me as i can still read it and it works fine